index.js

```diff
@@ -22,10 +23,16 @@ module.exports = function (opts) {
 			from: file.path,
 			to: file.path
 		}).then(function (res) {
+			var map;
 			file.contents = new Buffer(res.css);
 
 			if (res.map && file.sourceMap) {
-				applySourceMap(file, res.map.toString());
+				map = res.map.toJSON();
+				map.file = file.relative;
+				map.sources = [].map.call(map.sources, function (source) {
+					return path.join(path.dirname(file.relative), source);
+				});
```

Comment:
It works without, but the reason `gulp-postcss` is doing it is that it's missing the path. If we don't do this, all files will look like they are coming from the root of the website.

I did it mostly due to `gulp-postcss` doing the same thing, and they are maintaining both `postcss` and `gulp-postcss` seeming that they think it's the right solution. But one can argue that the problem is postcss.

I can remove it if you like :-)